### PR TITLE
fix: fix experiment archive action going out of sync [DET-4535]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

- Fix experiment archive action button going out of sync.
- Fix experiment actions not displaying loading spinner.

## Commentary (optional)

if you use any of the action buttons in experiment details a new request is initiated to update the state of the experiment and that doesn't always reflect the correct state. running curl can verify the API blocks properly 
curl-da.sh http://SERVER:8080/api/v1/experiments/716/archive -X POST -d '{}' && curl-da.sh http://SERVERi:8080/experiments/716/summary | jq '.archived'

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/pages/ExperimentDetails/ExperimentActions.tsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ const ExperimentActions: React.FC<Props> = ({ experiment, onClick, onSettled }:
   });
 
   const handleArchive = useCallback((archive: boolean) => async (): Promise<void> => {
-    setButtonStates(state => ({ ...state, archive }));
+    setButtonStates(state => ({ ...state, archive: true }));
```
Comment (hamidzr) [inline webui/react/src/pages/ExperimentDetails/ExperimentActions.tsx:56]: buttonLoadingStates might be a better name

Comment (hkang1) [inline webui/react/src/pages/ExperimentDetails/ExperimentActions.tsx:56]: **block:** This is probably something I did, but `archive` and all the other properties of `setButtonStates` should be uppercase, or we can just change all the initial values to be lowercase (in lines 46-52)`.

Comment (hamidzr) [reply webui/react/src/pages/ExperimentDetails/ExperimentActions.tsx]: `btnLoadingState` ? we can consider `btn` a known shorthand IMO

**Review by hkang1 — APPROVED** [review]
LGTM! Thanks for fixing this so quickly!

**Pull request merged**